algorithm_rgb.py

```diff
@@ -1,62 +1,184 @@
-# all white space is kept intact; don't add any extra whitespace since it may cause name comparisons
-# to fail.
-# !! Replace the content of this string with your variable names
-VARIABLE_NAMES = 'size of image channels'
+ALGORITHM_AUTHOR = 'Chris Schnaufer, Clairessa Brown, David Lebauer'
+ALGORITHM_AUTHOR_EMAIL = 'schnaufer@arizona.edu, clairessabrown@email.arizona.edu, dlebauer@email.arizona.edu'
+ALGORITHM_CONTRIBUTORS = ["Jacob van der Leeuw"]
+
+ALGORITHM_NAME = 'Greenness Transformer'
+ALGORITHM_DESCRIPTION = 'This algorithm performs a variety of calculations using RGB pixels from images in order' \
+                        'to assess plant and crop health and growth'
+
+# Citation information for publication
+CITATION_AUTHOR = 'Clairessa Brown'
+CITATION_TITLE = 'Woebbecke, D.M. et al'
+CITATION_YEAR = '2020'
+
+# The name of one or more variables returned by the algorithm, separated by commas
+
+VARIABLE_NAMES = 'excess greenness index, green leaf index, cive, normalized difference index, excess red, ' \
+                 'exgr, combined indices 1, combined indices 2, vegetative index, normalized green-red difference,' \
+                 ' percent green'
 
 # Variable units matching the order of VARIABLE_NAMES, also comma-separated.
-# For each variable name in VARIABLE_NAMES add the unit of measurement the value represents.
-# !! Replace the content of this string with your variables' unit
-VARIABLE_UNITS = 'pixels'
+VARIABLE_UNITS = '[-510:510], [-1:1], [-255:255], [-127:129], [-255:255], [-255:332], ' \
+                 '[-1000:1000], [-1000:1000], [-255:255], [-255:255], [0:100]'
 
 # Variable labels matching the order of VARIABLE_NAMES, also comma-separated.
-# This is an optional definition and can be left empty.
-VARIABLE_LABELS = ''
+VARIABLE_LABELS = 'excess_greenness_index, green_leaf_index, cive, normalized_difference_index(pxarray), ' \
+                'excess_red, exgr, combined_indices_1, combined_indices_2, vegetative_index, ngrdi, percent_green'
 
 # Optional override for the generation of a BETYdb compatible csv file
-# Set to False to suppress the creation of a compatible file
 WRITE_BETYDB_CSV = True
 
 # Optional override for the generation of a TERRA REF Geostreams compatible csv file
-# Set to False to suppress the creation of a compatible file
 WRITE_GEOSTREAMS_CSV = True
 
 
 # Entry point for plot-level RBG algorithm
-def calculate(pxarray: np.ndarray):
+
+
+def excess_greenness_index(pxarray: np.ndarray) -> float:
+    """
+    Minimizes variation between different illuminations and enhances detection
+    of plants
+    """
+    red, green, blue = get_red_green_blue_averages(pxarray)
+
+    return round(2 * green - (red + blue), 2)
+
+
+def green_leaf_index(pxarray: np.ndarray) -> float:
+    """
+    Calculates the green leaf index of an image
+    """
+    red, green, blue = get_red_green_blue_averages(pxarray)
+
+    return round((2 * green - red - blue) / (2 * green + red + blue), 2)
```

Comment:
Should check for zero denominator and return zero so that an exception isn't thrown. Here and below

Reply:
updated and tested with all black and all white GIMP images